### PR TITLE
Move queue to gpu when resuming checkpoint - SWAV self supervised model

### PR DESCRIPTION
**Pull request by thiyagu145**
## What does this PR do?

Fixes # (issue)

## Before submitting
- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [X] Did you read the [contributor guideline](https://github.com/PyTorchLightning/lightning-bolts/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [X] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests? [not needed for typos/docs]
- [X] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/lightning-bolts/blob/master/CHANGELOG.md)?

## PR review
 - [x] Is this pull request ready for review? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by ananyahjha93 — APPROVED** [review]

**Comment by ananyahjha93** [comment]
@thiyagu145 good catch. thanks for the PR.

File: pl_bolts/models/self_supervised/swav/swav_module.py [hunk 1]
```diff
@@ -183,8 +183,8 @@ def on_train_epoch_start(self):
                     self.feat_dim,
                 )
 
-                if self.gpus > 0:
-                    self.queue = self.queue.cuda()
+            if self.gpus > 0 and self.queue is not None:
+                self.queue = self.queue.cuda()
```
Comment (awaelchli) [inline pl_bolts/models/self_supervised/swav/swav_module.py:187]:
```suggestion
            self.queue is not None:
                self.queue = self.queue.to(self.device)
```
this is device agnostic, no need to check the gpus argument.

**Review by awaelchli — APPROVED** [review]

**Comment by Borda** [comment]
@thiyagu145 @ananyahjha93 mind check the failing tests?

**Comment by thiyagu145** [comment]
hi @Borda 
```TypeError: transfer_batch_to_device() missing 1 required positional argument: 'dataloader_idx'```
I'm not sure if this error is coming up because of the usage of self.queue.to(device).

**Review by ethanwharris — APPROVED** [review]
LGTM :smiley:

**Pull request merged**